odl/solvers/nonsmooth/proximal_operators.py

```diff
@@ -248,14 +262,11 @@ def proximal_arg_scaling(prox_factory, scaling):
     2011.
     """
 
-    if scaling.imag != 0:
-        raise ValueError("complex scaling not supported.")
-    else:
-        scaling = float(scaling.real)
-    if scaling == 0:
+    if scaling.imag != 0 * scaling:
```

Comment:
If scaling is a list, this gives

```
0 * [1, 2, 3] = []
```

etc. So the list has to be converted to a vector I guess?

Reply:
Hmmm, the problem here is that for a SeparableSum functional, the “canonical” choice for a stepsize would be a list, which cannot be converted to a vector.

Reply:
I would not check this at all. You also allow `scaling` to be a space element, and checking this can be quite expensive. Also this check will break down for arrays with the message
```
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

Reply:
But if you like you can also do a check only if you have "small stuff", like a scalar or a short list of scalings. In that case the documentation should mention this "inconsistency for performance reasons".

Reply:
I removed all the checks, except for those stuff which satisfies np.isscalar. The latter is necessary to avoid regressions, because in the master branch, scaling with zero returned the identity operator.